packages/blade/src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,204 @@
+   * Error text of the checkbox
+   *
+   * Renders when `validationState` is set to 'error'
+   *
+   * Overrides helpText
+   */
+  errorText?: string;
+  /**
+   * If `true`, the checkbox will be indeterminate.
+   * This does not modify the isChecked property.
+   *
+   * @default false
+   */
+  isIndeterminate?: boolean;
+  /**
+   * The name of the input field in a checkbox
+   * (Useful for form submission).
+   */
+  name?: string;
+  /**
+   * The value to be used in the checkbox input.
+   * This is the value that will be returned on form submission.
+   */
+  value?: string;
+  /**
+   * If `true`, the checkbox will be disabled
+   *
+   * @default false
+   */
+  isDisabled?: boolean;
+  /**
+   * If `true`, the checkbox input is marked as required,
+   * and `required` attribute will be added
+   *
+   * @default false
+   */
+  isRequired?: boolean;
+  /**
+   * If `error`, the checkbox input is marked as invalid,
+   * and `invalid` attribute will be added
+   *
+   * @default undefined
+   */
+  validationState?: 'error' | undefined;
+};
+
+const Checkbox = ({
+  defaultChecked,
+  validationState,
+  isChecked,
+  isDisabled,
+  isIndeterminate,
+  isRequired,
+  name,
+  onChange,
+  value,
+  children,
+  helpText,
+  errorText,
+}: CheckboxProps): React.ReactElement => {
+  const groupProps = useCheckboxGroupContext();
+
+  // ban controlling checkbox manually while inside group
+  const hasDefaultChecked = !isUndefined(defaultChecked);
+  const hasIsChecked = !isUndefined(isChecked);
+  const hasOnChange = !isUndefined(onChange);
+  if ((hasDefaultChecked || hasIsChecked || hasOnChange) && !isEmpty(groupProps)) {
+    const props = [
+      hasDefaultChecked ? 'defaultChecked' : undefined,
+      hasIsChecked ? 'isChecked' : undefined,
+      hasOnChange ? 'onChange' : undefined,
+    ].filter(Boolean);
+
+    throw new Error(
+      `[Blade Checkbox]: Cannot set \`${props.join(
+        ',',
+      )}\` on <Checkbox /> when it's inside <CheckboxGroup />, Please set it on the <CheckboxGroup /> itself`,
+    );
+  }
+  // mandate value prop when using inside group
+  if (!value && !isEmpty(groupProps)) {
+    throw new Error(
+      `[Blade Checkbox]: <CheckboxGroup /> requires that you pass unique "value" prop to each <Checkbox />
+      <CheckboxGroup>
+        <Checkbox value="apple">Apple</Checkbox>
+        <Checkbox value="mango">Mango</Checkbox>
+      </CheckboxGroup>
+      `,
+    );
+  }
+  // If validationState is used while inside group
+  if (validationState && !isEmpty(groupProps)) {
+    throw new Error(
+      "[Blade Checkbox]: Can't use validationState of individual Checkboxes inside <CheckboxGroup />, please set validtionState on the <CheckboxGroup /> itself.",
+    );
+  }
+
+  const _validationState = validationState ?? groupProps.validationState;
+  const _hasError = _validationState === 'error';
+  const _isDisabled = isDisabled ?? groupProps.isDisabled;
+  const _name = name ?? groupProps.name;
```

Comment:
what if I provide name to individual checkbox while using them under checkboxgroup?

Reply:
Consumers can. maybe they have a group but still want to split the form values to different sections before sending it to server. 

```ts
<CheckboxGroup>
  <Checkbox name="fruits">apple</Checkbox>
  <Checkbox name="fruits">mango</Checkbox>
  <Checkbox name="flowers">sunflower</Checkbox>
  <Checkbox name="flowers">dandelion</Checkbox>
</CheckboxGroup>
```

Reply:
Then we can make it so that inside group name won't work.

Reply:
Done.